### PR TITLE
Update Akka and Akka Streams. Will fix potential memory leak. 

### PR DESCRIPTION
**Pull request by guizmaii**
More info: https://twitter.com/akkateam/status/939063569953341440

### CODE REVIEW

**Comment by guizmaii** [comment]
I'm resolving the compilation error

File: dynamodb/src/main/scala/akka/stream/alpakka/dynamodb/impl/DynamoClientImpl.scala [hunk 1]
```diff
@@ -22,12 +24,12 @@ class DynamoClientImpl(
   override protected val service = "dynamodb"
   override protected val defaultContentType =
     ContentType.Binary(MediaType.customBinary("application", "x-amz-json-1.0", NotCompressible))
-  override protected implicit val ec = system.dispatcher
+  override protected implicit val ec: ExecutionContextExecutor = system.dispatcher
 
   override protected val connection: AwsConnect =
     if (settings.port == 443)
-      Http().cachedHostConnectionPoolHttps[AwsRequestMetadata](settings.host)(materializer)
+      Http().cachedHostConnectionPoolHttps(settings.host)
```
Comment (guizmaii) [inline dynamodb/src/main/scala/akka/stream/alpakka/dynamodb/impl/DynamoClientImpl.scala]: Explicitly passing the `materializer` here makes the compilation fails. It's strange !

File: dynamodb/src/main/scala/akka/stream/alpakka/dynamodb/impl/DynamoClientImpl.scala [hunk 2]
```diff
@@ -22,12 +24,12 @@ class DynamoClientImpl(
...
     else
-      Http().cachedHostConnectionPool[AwsRequestMetadata](settings.host, settings.port)(materializer)
+      Http().cachedHostConnectionPool[AwsRequestMetadata](settings.host, settings.port)
```
Comment (guizmaii) [inline dynamodb/src/main/scala/akka/stream/alpakka/dynamodb/impl/DynamoClientImpl.scala]: Same here

File: dynamodb/src/main/scala/akka/stream/alpakka/dynamodb/impl/DynamoClientImpl.scala [hunk 3]
```diff
@@ -22,12 +24,12 @@ class DynamoClientImpl(
   override protected val service = "dynamodb"
   override protected val defaultContentType =
     ContentType.Binary(MediaType.customBinary("application", "x-amz-json-1.0", NotCompressible))
-  override protected implicit val ec = system.dispatcher
+  override protected implicit val ec: ExecutionContextExecutor = system.dispatcher
 
   override protected val connection: AwsConnect =
     if (settings.port == 443)
-      Http().cachedHostConnectionPoolHttps[AwsRequestMetadata](settings.host)(materializer)
+      Http().cachedHostConnectionPoolHttps[AwsRequestMetadata](settings.host)
```
Comment (guizmaii) [inline dynamodb/src/main/scala/akka/stream/alpakka/dynamodb/impl/DynamoClientImpl.scala]: Explicitly passing the `materializer` here makes the compilation fails. It's strange !

File: dynamodb/src/main/scala/akka/stream/alpakka/dynamodb/impl/DynamoClientImpl.scala [hunk 4]
```diff
@@ -22,12 +24,12 @@ class DynamoClientImpl(
...
     else
-      Http().cachedHostConnectionPool[AwsRequestMetadata](settings.host, settings.port)(materializer)
+      Http().cachedHostConnectionPool[AwsRequestMetadata](settings.host, settings.port)
```
Comment (guizmaii) [inline dynamodb/src/main/scala/akka/stream/alpakka/dynamodb/impl/DynamoClientImpl.scala]: Same here

File: dynamodb/src/main/scala/akka/stream/alpakka/dynamodb/impl/DynamoClientImpl.scala (hunk 3 above)
Comment (johanandren) [reply dynamodb/src/main/scala/akka/stream/alpakka/dynamodb/impl/DynamoClientImpl.scala]: It's a known binary incompability/regression: https://github.com/akka/akka-http/issues/1603

**Comment by guizmaii** [comment]
Commits squashed ;)

**Comment by johanandren** [comment]
(Note that we can squash through the GitHub UI so no need in general to do that)

**Comment by ennru** [comment]
Thank you for this contribution. Keep them coming!

**Pull request merged**